### ISSUE TITLE
(typescript) TS2430: Interface 'NumberFormatProps' incorrectly extends interface 'InputHTMLAttributes<HTMLInputElement>'.

### ISSUE THREAD

**Issue by deadNightTiger** [s-yadav/react-number-format#244]
```
ERROR in .../node_modules/react-number-format/typings/number_format.d.ts(23,20):
TS2430: Interface 'NumberFormatProps' incorrectly extends interface 'InputHTMLAttributes<HTMLInputElement>'.
  Types of property 'defaultValue' are incompatible.
    Type 'string | number | undefined' is not assignable to type 'string | string[] | undefined'.
      Type 'number' is not assignable to type 'string | string[] | undefined'.
```
Basically `InputHTMLAttributes` declares `defaultValue` to be either `string`, `string[]`, or `undefined`, and `number` is neither of them.

**Comment by ryantxu** [comment]
I get the same thing, forcing to "3.6.1" works though

**Comment by s-yadav** [comment]
@deadNightTiger, @ryantxu  Fixed on 4.0.1. Let me know if it works

**Comment by ryantxu** [comment]
works for me!  Thanks
`✨  Done in 17.83s.`
